keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/Stepsize.java

```diff
@@ -0,0 +1,165 @@
+package io.improbable.keanu.algorithms.mcmc;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.ProbabilityCalculator;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexId;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.diff.LogProbGradientCalculator;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Used by NUTS as an epsilon for the leap frog
+ */
+public class Stepsize {
+
+    private static final double STABILISER = 10;
+    private static final double SHRINKAGE_FACTOR = 0.05;
+    private static final double TEND_TO_ZERO_EXPONENT = 0.75;
+    private static final double STARTING_STEPSIZE = 1;
+
+    private double stepsize;
+    private double averageTreeAcceptanceProb;
+    private double averageAcceptanceProb;
+    private double targetAcceptanceProb;
+    private double logStepSizeFrozen;
+    private double adaptCount;
+    private double shrinkageTarget;
+
+    /**
+     * @param stepsize                  the step size
+     * @param targetAcceptanceProb      the target acceptance probability (lower target equates to a higher step size when tuning)
+     * @param adaptCount                the number of samples to adapt for
+     */
+    Stepsize(double stepsize, double targetAcceptanceProb, int adaptCount) {
+        this.averageAcceptanceProb = 0.;
+        this.targetAcceptanceProb = targetAcceptanceProb;
+        this.stepsize = stepsize;
+        this.logStepSizeFrozen = Math.log(stepsize);
+        this.adaptCount = adaptCount;
+        this.shrinkageTarget = Math.log(10 * stepsize);
```

Comment:
`shrinkageTarget` never changes - you could implement it as a private method instead of a field

Reply:
Oh OK. Carry on.